pkg/external-provider/errors.go

```diff
@@ -0,0 +1,21 @@
+package provider
+
+import "errors"
+
+var (
+	// ErrNewOperatorNotSupportedByPrometheus creates an error that represents the fact that we were requested to service a query that
+	// Prometheus would be unable to support.
+	ErrNewOperatorNotSupportedByPrometheus = errors.New("operator not supported by prometheus")
+
+	// ErrNewOperatorRequiresValues creates an error that represents the fact that we were requested to service a query
+	// that was malformed in its operator/value combination.
+	ErrNewOperatorRequiresValues = errors.New("operator requires values")
```

Comment:
nit: doesn't `ErrMalformedQuery` make more sense here?

Reply:
Your phrasing like the one above points people at the query. Should we have more universal query error and consolidate? I'm happy either way.